### PR TITLE
WIP: NFTs burn problem

### LINKED ISSUE #921: Fe: NFT burns after cancelling
- User selects NFT from their inventory
- Hits transfer to mainnet
- Sees the crazy gas fees, cancels
- NFT no longer in their inventory

### PR DESCRIPTION
**Pull request by avaer**
Fixes https://github.com/webaverse/app/issues/921

### CODE REVIEW

File: functions/AssetFunctions.js [hunk 1]
```diff
@@ -137,15 +137,72 @@ export const getStuckAsset = async (tokenName, tokenId, globalState) => {
 
   let depositedFiltered;
   if (tokenId) {
-    depositedFiltered = depositedEntries.filter(entry => entry.returnValues[0].toLowerCase() === address && entry.returnValues[1] === tokenId.toString());
+    depositedFiltered = depositedEntries.filter(entry => entry.returnValues[0].toLowerCase() === mainnetAddress ? mainnetAddress : address && entry.returnValues[1] === tokenId.toString());
```
Comment (avaer) [inline functions/AssetFunctions.js:140]: I don't understand why this change is here -- it seems we would need two cases, one for getting stuck on the way to mainnet and another getting stuck on the way to the sidechain.

Is this code handling that?

**Comment by avaer** [comment]
We need to make sure that both directions of transfer are being handled.

This might involve having to bootstrap the Rinkeby network again, so we can test that the logic works without paying fees.

File: webaverse/blockchain.js [hunk 2]
```diff
@@ -154,7 +155,8 @@ const runSidechainTransaction = mnemonic => async (contractName, method, ...args
     common,
   }).sign(privateKeyBytes);
   const rawTx = '0x' + tx.serialize().toString('hex');
-  const receipt = await web3['back'].eth.sendSignedTransaction(rawTx);
+  const receipt = await web3['back'].eth.sendSignedTransaction(rawTx, function(err, hash){
```
Comment (avaer) [inline webaverse/blockchain.js]: This sounds like it would lock up? Since you provide both a callback and an await?

File: webaverse/blockchain.js [hunk 3]
```diff
@@ -135,9 +136,9 @@ const runSidechainTransaction = mnemonic => async (contractName, method, ...args
 
   const txData = contracts['back'][contractName].methods[method](...args);
   const data = txData.encodeABI();
-  const gas = await txData.estimateGas({
-    from: address,
-  });
+  // const gas = await txData.estimateGas({
```
Comment (avaer) [inline webaverse/blockchain.js]: Yeah not needed.

File: webaverse/blockchain.js [hunk 4]
```diff
@@ -5,6 +5,7 @@ const hdkey = hdkeySpec.default;
 import ethereumJsTx from '../libs/ethereumjs-tx.js';
 import {makePromise} from './util.js';
 import {storageHost, web3MainnetSidechainEndpoint, web3RinkebySidechainEndpoint} from './constants.js';
+import { LoadingManager } from '../libs/three.module.js';
```
Comment (avaer) [inline webaverse/blockchain.js]: This is not used.

File: functions/UIStateFunctions.js [hunk 5]
```diff
@@ -101,8 +101,7 @@ export const getToken = async (id, hostname) => {
     const { getNetworkName } = await getBlockchain();
     networkName = getNetworkName();
   }
-
-  const res = await fetch(`${networkName !== "main" ? `https://mainnetall-tokens.webaverse.com/${id}` : `https://mainnetall-tokens.webaverse.com/${id}`}`);
+  const res = await fetch(`${networkName !== "main" ? `https://rinkebysidechain-tokens.webaverse.com/${id}` : `https://mainnetall-tokens.webaverse.com/${id}`}`);
```
Comment (avaer) [inline functions/UIStateFunctions.js]: Should this be using a "rinkeby all" endpoint?

File: webaverse/blockchain.js (hunk 2 above)
Comment (charisra) [reply webaverse/blockchain.js]: This can be reverted to how it was. I mainly added the callback to help me debug. Not needed anymore.

File: webaverse/blockchain.js (hunk 4 above)
Comment (charisra) [reply webaverse/blockchain.js]: There is a ton of unused imports across the codebase. I’m removing those as I’m seeing. Plus unused variables, calls (some of which caused errors or bugs), like the gas one above

Comment (avaer) [reply webaverse/blockchain.js]: But this one adds an unused import.

File: webaverse/blockchain.js (hunk 2 above)
Comment (avaer) [reply webaverse/blockchain.js]: How does the callback help to debug when the async is the same thing?

File: webaverse/blockchain.js (hunk 4 above)
Comment (charisra) [reply webaverse/blockchain.js]: Right, this new line can be removed

File: webaverse/blockchain.js (hunk 2 above)
Comment (avaer) [reply webaverse/blockchain.js]: Why not 
```
let receipt;
try {
receipt = await web3['back'].eth.sendSignedTransaction(rawTx);
} catch (err) {
console.warn(err);
}

Comment (avaer) [reply webaverse/blockchain.js]: The problem with the callback is it might actually prevent the promise from working, since having callbacks and promises together would be a bad code style and libraries might work on that understanding.

So it might just not be a promise if you do that, thus breaking the code while you debug it.

I don't know what the facts are but this looks very fishy.

Comment (charisra) [reply webaverse/blockchain.js]: As it’s written now, it doesn’t. I had a console.log written, which I removed before committing. The line can be reverted, the callback is not used.

File: functions/UIStateFunctions.js (hunk 5 above)
Comment (charisra) [reply functions/UIStateFunctions.js]: Great! Thanks for that. This actually works and I can see my minted data getting returned properly, e.g. http://rinkebyall-tokens.webaverse.com/3

please change it to rinkeby all

Comment (avaer) [reply functions/UIStateFunctions.js]: If I take on the issue I'll change it.
